File: src/main/java/org/kite9/diagram/visualization/orthogonalization/Dart.java
package org.kite9.diagram.visualization.orthogonalization;

import java.util.Map;

import org.kite9.diagram.common.elements.edge.Edge;
import org.kite9.diagram.model.DiagramElement;
import org.kite9.diagram.model.position.Direction;

/**
 * Darts are created by the orthogonalization process.  They are directed edges within the diagram which 
 * represent the borders of Retangular diagram elements, or Connections.
 * 
 * @author robmoffat
 *
 */
public interface Dart extends Edge {

	Map<DiagramElement, Direction> getDiagramElements();
	
	public String getID();

}

File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/Dart.kt
package org.kite9.diagram.visualization.orthogonalization

import org.kite9.diagram.common.elements.edge.Edge
import org.kite9.diagram.model.DiagramElement
import org.kite9.diagram.model.position.Direction

/**
 * Darts are created by the orthogonalization process.  They are directed edges within the diagram which
 * represent the borders of Retangular diagram elements, or Connections.
 *
 * @author robmoffat
 */
interface Dart : Edge {

    fun getDiagramElements(): MutableMap<DiagramElement, Direction?>

    /**
     * Darts always have a direction
     */
    override fun getDrawDirection(): Direction
}